### PR TITLE
chore(ecs): add app autoscaling integ test for Amazon ECS createServerGroup

### PR DESCRIPTION
**Pull request by paragbhingre**
This PR contains integration test for Create Server Group Operation for the task definition input, EC2 launch type, and target group mappings with application auto scaling source / policies.

### CODE REVIEW

**Comment by paragbhingre** [comment]
Please review this PR @allisaurus

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        .thenReturn(
+            new PutScalingPolicyResult()
+                .withPolicyARN(
+                    "arn:aws:autoscaling:us-west-2:012345678910:scalingPolicy:6d8972f3-efc8-437c-92d1-6270f29a66e7:resource/ecs/service/default/web-app:policyName/web-app-cpu-gt-75"));
+
+    when(mockAmazonCloudWatchClient.describeAlarms(any(DescribeAlarmsRequest.class)))
+        .thenReturn(
+            new DescribeAlarmsResult()
+                .withMetricAlarms(
+                    Arrays.asList(new MetricAlarm().withAlarmName("TargetTracking-testAlarm"))));
```
Comment (allisaurus) [inline clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java:140]: this is the only place we refer to the test alarm as "TargetTracking-testAlarm" (everywhere else it's just "testAlarm"). I know we're not asserting on this mocked return value, but it was confusing to see assertions on the `DescribeAlarmsResults` args below succeeding with a different name. Can we please use one or the other everywhere?

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        "integInputsEc2TargetGroupMappingsWithAutoScaling-cluster",
+        seenCreateServRequest.getCluster());
+
+    ArgumentCaptor<PutScalingPolicyRequest> scalableTargetsRequestArgumentCaptor =
+        ArgumentCaptor.forClass(PutScalingPolicyRequest.class);
+    verify(mockAWSApplicationAutoScalingClient)
+        .putScalingPolicy(scalableTargetsRequestArgumentCaptor.capture());
+    PutScalingPolicyRequest seenDescribeScalableTargetsRequest =
+        scalableTargetsRequestArgumentCaptor.getValue();
+    assertEquals("createdServiceTestPolicy", seenDescribeScalableTargetsRequest.getPolicyName());
```
Comment (allisaurus) [inline clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java:237]: what mechanism is appending `TestPolicy` to the service name? something in clouddriver or in AWS?

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    assertEquals(expectedServerGroupName + "-v000", seenCreateServRequest.getServiceName());
+    assertEquals(1, seenCreateServRequest.getLoadBalancers().size());
+    LoadBalancer serviceLB = seenCreateServRequest.getLoadBalancers().get(0);
+    assertEquals("v000", serviceLB.getContainerName());
+    assertEquals(80, serviceLB.getContainerPort().intValue());
+    assertEquals(
+        "integInputsEc2TargetGroupMappingsWithAutoScaling-cluster",
+        seenCreateServRequest.getCluster());
+
+    ArgumentCaptor<PutScalingPolicyRequest> scalableTargetsRequestArgumentCaptor =
```
Comment (allisaurus) [inline clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java:231]: call this var `putScalingPolicyRequestArgumentCaptor` to reflect the request obj. "scalableTargetsRequest" is ambiguous given we're calling `describeScalableTargets` as well.

Comment (allisaurus) [reply clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java:231]: ^ also:  doesn't seem like we're asserting on the args sent to `DescribeScalableTargets` or `DescribeScalingPolicies` - is there value on asserting on their values?

**Review by allisaurus — COMMENTED** [review]
Thanks, @paragbhingre ! A couple comments re: terminology & questions, but no major concerns.

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java (hunk 2 above)
Comment (paragbhingre) [reply clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java:237]: Hey at this particular [place](https://github.com/spinnaker/clouddriver/blob/6d6831654b85e6955b67c15e84e313fa99da1796/clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/services/EcsCloudMetricService.java#L340-L341) `createService` gets concatenated with `TestPolicy`. Does that solve your concern?

**Comment by allisaurus** [comment]
@paragbhingre can you also please also change the commit prefix to "chore(ecs)" so it doesn't show up in future changelogs as a feature? I think this'll be a good convention to keep for any future integ tests.

**Comment by paragbhingre** [comment]
> @paragbhingre can you also please also change the commit prefix to "chore(ecs)" so it doesn't show up in future changelogs as a feature? I think this'll be a good convention to keep for any future integ tests.

Hey @allisaurus I am making sure that I will prefix "chore(ecs)" to all the new commits. I will merge the review comments and change the commit message as well (with chore(ecs)).

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java (hunk 1 above)
Comment (paragbhingre) [reply clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java]: @allisaurus I have made sure to keep the name consistent i.e. "testAlarm" everywhere.

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java (hunk 3 above)
Comment (paragbhingre) [reply clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java]: There is not much to assert on DescribeScalableTargets or DescribeScalingPolicies, but I have added couple asserts on them in the recent commit.

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        .thenReturn(mockELB);
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given description w/ inputs, EC2 launch type, and target group mappings "
+          + "with Application Auto Scaling, successfully submit createServerGroup operation"
+          + "\n===")
+  @Test
+  public void createServerGroup_InputsEc2TargetGroupMappingsWithAutoScalingGroupTest()
```
[the remark this answers is not in the document]
Comment (paragbhingre) [reply clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithAppAutoScalingSpec.java]: Resolved

**Review by allisaurus — APPROVED** [review]

**Pull request merged**